### ISSUE TITLE
The auto-generated text for spellcasting should be editable

### ISSUE THREAD

**Issue by DrYoshiyahu** [zeel01/MonsterBlocks#117]
This is purely aesthetic, but I'm sure one would agree that named characters should not be referred to in the same way that unnamed monsters are. The problem, specifically, is that a character called "John" will have a spellcasting feature that reads

"**The John** is an nth-level spellcaster. **Its** spellcasting ability is X (...) **The **John** has the following spells prepared:"

Obviously "The" should be completely removable, if the user desires, and "its" should be changed to "he" in this case, but "she" and "they" would also be valid options.

The name itself should also be editable. If the creature is called "John, the Destroyer of Worlds and Harbinger of Death," the user may only want "John" in the feature descriptions, to save space.

Also, it gets *really* ugly if NPCs have naming conventions for sorting, such as `last name, first name`. Then you get "The Doe, John is an nth-level spellcaster," which is horrifying.

**Comment by zeel01** [comment]
Yeah, this is also an issue with legendary actions:

![image](https://user-images.githubusercontent.com/1721836/132969681-376098c5-7713-4236-b519-d5d4868d8075.png)

It should just say "The dragon can..." but it doesn't know that there is a short-form of the name it could use. If this was the stat block for "Strahd von Zarovich" it should probably say "Strahd" or "The vampire" but not "Strahd von Zarovich" and certainly not "The Strahd von Zarovich."

However, there is a problem: There isn't anywhere to store this data. Okay, not rally true, it could go in a flag but... I hate that. I'm not a fan of sheets adding extra data fields to actors, it's one of the reasons there wasn't an AC label before the system added it. On top of that, there has to be an intuitive way to edit that value which adds a lot of complexity.

I'm not say "no, never" but it's a very low priority.

**Comment by DrYoshiyahu** [comment]
Why not just allow the entire textfield to be editable as plain text?

**Comment by zeel01** [comment]
It's not plain text though, it has multiple fill-ins including the caster level, certain spells, the spellcasting ability, etc. It's not just the description of the spellcasting ability item, it's a fully generated text based on the various attributes of the Actor.

**Comment by DrYoshiyahu** [comment]
No, I know, but why not give the user the ability to ignore that and just write whatever they want?

For example, the written description of the feature could override the premade one.

**Comment by zeel01** [comment]
### Manual Spellcasting Feature Descriptions
When this option is enabled, the description of a spellcasting feature item will replace the paragraph of text at the top of the feature before the spell list.

When enabled, the sheet will attempt to match up the spellcasting feature items on the sheet with the spellcasting features it created per #92. If a match is found, the description is subbed in instead of the generated text. If no matching feature is found, the generated text is used.
